src/TemplateHelpers.hs

```diff
@@ -125,7 +128,18 @@ renderReadme = \case
 renderHtmlDocs :: D.VerifiedPackage -> Text -> Handler (Maybe Html)
 renderHtmlDocs pkg mnString = do
   htmlRenderContext <- getHtmlRenderContext
-  let docsOutput = packageAsHtml (htmlRenderContext pkg) pkg
+
+  depHtmlRenderContexts <- do
```

Comment:
I think it's simpler to split off a top-level function actually, because then when you're reading the code you can immediately know that this function doesn't depend on extra stuff in scope inside the other function, such as bound variables for its arguments and other stuff in `where` clauses.